support-workers/src/typescript/services/config.ts

```diff
@@ -0,0 +1,54 @@
+import { GetParameterCommand, SSMClient } from '@aws-sdk/client-ssm';
+import type { z } from 'zod';
+import type { Stage } from '../model/stage';
+import { awsConfig } from '../util/awsConfig';
+
+export async function getConfig<I, O, T extends z.ZodType<O, z.ZodTypeDef, I>>(
+	stage: Stage,
+	configKeyName: string,
+	schema: T,
+): Promise<O> {
```

Comment:
why `<I, O, T extends z.ZodType<O, z.ZodTypeDef, I>>`
with `schema: T,`
and not `<O>`
with `schema: z.ZodType<O, z.ZodTypeDef, any>`
?  ( @tjmw and I were looking at similar code and wondering this morning )